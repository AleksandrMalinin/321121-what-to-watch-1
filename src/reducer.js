import {films} from './mocks/films.js';
import {constants} from './constants.js';

const initialState = {
  moviesList: films,
  activeGenre: constants.DEFAULT_GENRE
};

const ActionCreators = {
  changeGenre: (genre) => {
    return {
      type: constants.ACTION_TYPE.changeFilterGenre,
      payload: genre
    };
  },

  getMoviesList: (genre) => {
    let movies = genre === initialState.activeGenre ? initialState.moviesList.slice(1) : initialState.moviesList.filter((film) => genre === film.genre);

    return {
      type: constants.ACTION_TYPE.getMoviesList,
      payload: movies
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case constants.ACTION_TYPE.changeFilterGenre: return Object.assign({}, state, {
      activeGenre: action.payload
    });

    case constants.ACTION_TYPE.getMoviesList: return Object.assign({}, state, {
      moviesList: action.payload
    });
  }

  return state;
};

export {reducer, ActionCreators};
