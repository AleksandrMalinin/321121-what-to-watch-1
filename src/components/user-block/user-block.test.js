import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {createStore} from 'redux';
import reducer from '../../reducer';
import UserBlock from './user-block.jsx';

const mockStore = createStore(reducer);

it(`UserBlock correctly renders`, () => {
  const tree = renderer
  .create(
      <Provider store={mockStore}>
        <BrowserRouter>
          <UserBlock
            isAuthorizationRequired={true}
          />
        </BrowserRouter>
      </Provider>)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
