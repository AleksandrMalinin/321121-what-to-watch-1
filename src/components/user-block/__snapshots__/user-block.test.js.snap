// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock correctly renders 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <img
        alt=""
        height="63"
        src=""
        width="63"
      />
    </a>
  </div>
</div>
`;
